### PR TITLE
Fixed issue #18464: Min/max settings doesn't work for question attrib…

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue #18464: Min/max settings doesn't work for question attrib…ute with type of integer
Dev: add the HTML attribute (clearly removed from code)

### CODE REVIEW

File: application/models/QuestionAttribute.php [hunk 1]
```diff
@@ -380,6 +380,8 @@ public static function getDefaultSettings()
             "readonly_when_active" => false,
             "expression" => null,
             "xssfilter" => true,
+            "min" => null,
```
Comment (gabrieljenik) [inline application/models/QuestionAttribute.php:383]: I am not sure about adding these here. Why?

File: application/extensions/AdvancedSettingWidget/views/integer.php [hunk 2]
```diff
@@ -4,17 +4,15 @@
...
-        min=""
-        -->
     <input 
         type="number" 
         class="form-control" 
         name="<?= $inputBaseName ?>"
         id="<?= CHtml::getIdByName($inputBaseName); ?>"
         <?= ($this->setting['help']) ? 'aria-describedby="help-' . CHtml::getIdByName($inputBaseName) . '"' : "" ?>
         value="<?= CHtml::encode($this->setting['value']); ?>"
+        <?= isset($this->setting['min']) ? 'min="' . $this->setting['min'] . '"' : '' ?>
```
Comment (gabrieljenik) [inline application/extensions/AdvancedSettingWidget/views/integer.php:14]: Checking like this is OK or should also check there is content?

**Review by gabrieljenik — APPROVED** [review]
Code looks OK generally speaking.
Please also recall to update manual

File: application/extensions/AdvancedSettingWidget/views/integer.php (hunk 2 above)
Comment (Shnoulle) [reply application/extensions/AdvancedSettingWidget/views/integer.php:14]: What do you mean ?

**Comment by Shnoulle** [comment]
>  Please also recall to update manual

This work in 3.LTS , not in 5, but right : nothing in manual for min/max.
About set it to null in default value : i really think it's more clean and clear for other dev.

**Comment by Shnoulle** [comment]
>  Please also recall to update manual

Must add XSS (and the complete version number since)

File: application/models/QuestionAttribute.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/models/QuestionAttribute.php:383]: Yes, that could help

File: application/extensions/AdvancedSettingWidget/views/integer.php (hunk 2 above)
Comment (Shnoulle) [reply application/extensions/AdvancedSettingWidget/views/integer.php:14]: > if the min setting contains a space, should the attribute be used or not?

If it's not an integer :). 
It's PHP or config.xml file : it's not user entered data. I think we don't have to fix it.

`value="<?= CHtml::encode($this->setting['value']); ?>"` don't check if it's integer here too :)

**Comment by Shnoulle** [comment]
> > @olleharstedt some core attribute need it i think.
> 
> Sure, but still waiting for test?

Adding unit or functional test do the trick when i really need a fix ? Or i add to hire a tester ?

**Comment by olleharstedt** [comment]
> > > @olleharstedt some core attribute need it i think.
> > 
> > 
> > Sure, but still waiting for test?
> 
> Adding unit or functional test do the trick when i really need a fix ? Or i add to hire a tester ?

Gabriel has a tester and we're trying to hire one too.

Yeah, automated tests could be an alternative, maybe. Do we have test instructions?

File: application/extensions/AdvancedSettingWidget/views/integer.php (hunk 2 above)
Comment (olleharstedt) [reply application/extensions/AdvancedSettingWidget/views/integer.php:14]: Use `??` here please.

**Review by olleharstedt — CHANGES_REQUESTED** [review]

File: application/extensions/AdvancedSettingWidget/views/integer.php (hunk 2 above)
Comment (olleharstedt) [reply application/extensions/AdvancedSettingWidget/views/integer.php:14]: Ah wait, not possible. Sorry, my bad.

Comment (Shnoulle) [reply application/extensions/AdvancedSettingWidget/views/integer.php:14]: In 3.X : we have a system to be sure to have something : https://github.com/LimeSurvey/LimeSurvey/blob/06e252130069f697b247446ae52c6b8b18d261d8/application/models/QuestionAttribute.php#L312

But in 5.X … no system

**Pull request merged**